sql/parser/builtins.go

```diff
@@ -248,6 +248,33 @@ var builtins = map[string][]builtin{
 		return DString(strings.Replace(s, from, to, -1)), nil
 	}, DummyString)},
 
+	"translate": {stringBuiltin3(func(s, from, to string) (Datum, error) {
+		const deletionRune = rune(0)
+		translation := make(map[rune]rune, len(from))
+		toLoc := 0
+		for _, fromRune := range from {
+			toRune, size := utf8.DecodeRuneInString(to[toLoc:])
+			if toRune == utf8.RuneError {
+				toRune = deletionRune
+			} else {
+				toLoc += size
```

Comment:
Since you're never accessing `to` after this loop, rather the tracking `toLoc` you could do: `to = to[size:]`.